File: itens/arma.js
import Item from './item.js';

export default class Arma extends Item {
  constructor(nome, danoMin, danoMax, descricao) {
    super(nome, descricao);
    this.danoMin = danoMin;
    this.danoMax = danoMax;
  }

  dano() {
    return Math.floor(Math.random() * (this.danoMax - this.danoMin + 1)) + this.danoMin;
  }
}

File: itens/arma.py
# itens/arma.py
from .item import Item

class Arma(Item):
    def __init__(self, nome: str, descricao: str = '', bonus_ataque: int = 0):
        super().__init__(nome, descricao)
        self.bonus_ataque = bonus_ataque

    def __repr__(self):
        return f"<Arma {self.nome} (+{self.bonus_ataque})>"
